openmc/model/surface_composite.py

```diff
@@ -52,6 +53,92 @@ def __neg__(self):
         """Return the negative half-space of the composite surface."""
 
 
+class CylinderSector(CompositeSurface):
+    """Infinite cylindrical sector composite surface
+
+    This class acts as a proper surface, meaning that unary `+` and `-`
+    operators applied to it will produce a half-space. The negative
+    side is defined to be the region inside of the cylinder sector.
+
+    Parameters
+    ----------
+    center : iterable of float
+       Coordinate for central axes of cylinders in the (y, z), (z, x), or
+         (x, y) basis. Defaults to (0,0)
+    r1 : float
+        Inner cylinder radii
+    r2 : float
+        Outer cylinder radii
+    theta0 : float
+        Angular offset of the sector in degrees relative to the primary basis
+        axis (+y, +z, or +x).
+    theta : float
+        Angular width of the sector in degrees.
+    axis : {'x', 'y', 'z'}
+        Central axis of the cylinders. Defaults to 'z'.
+    **kwargs
+        Keyword arguments passed to underlying plane classes.
+
+    Attributes
+    ----------
+    outer_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Outer cylinder surface
+    inner_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Inner cylinder surface
+    plane0 : openmc.Plane
+        Plane at angle :math:`\\theta_0` relative to the first basis axis
+    plane1 : openmc.Plane
```

Comment:
If adopting above suggestion regarding theta, `plane1` and `plane2` might be more intuitive (i.e., plane1 matches theta1, and plane2 matches theta2)